tests/load/locustfile.py

```diff
@@ -0,0 +1,405 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+import base64
+import json
+import time
+import uuid
+from contextlib import closing
+from urllib.parse import urljoin, urlparse
+
+from locust import HttpUser, TaskSet, events, task
+from websocket import create_connection
+from websocket._exceptions import WebSocketTimeoutException
+
+
+@events.init_command_line_parser.add_listener
+def _(parser):
+    parser.add_argument(
+        "--websocket_url",
+        type=str,
+        env_var="AUTOPUSH_WEBSOCKET_URL",
+        default="",
+        help="Server URL",
+    )
+    parser.add_argument(
+        "--endpoint_url",
+        type=str,
+        env_var="AUTOPUSH_ENDPOINT_URL",
+        default="",
+        help="Endpoint URL",
+    )
+
+
+class ConnectionTaskSet(TaskSet):
+    @task
+    def test_basic(self):
+        encrypted_data = "aLongStringOfEncryptedThings"
+        headers = {"TTL": "60", "Content-Encoding": "aes128gcm"}
+        channel_id = str(uuid.uuid4())
+        ws = create_connection(
+            self.user.environment.parsed_options.websocket_url,
+            header={"Origin": "http://localhost:1337"},
+            ssl=False,
+        )
+
+        body = json.dumps(dict(messageType="hello", use_webpush=True))
+        ws.send(body)
+        res = json.loads(ws.recv())
+        assert res["messageType"] == "hello"
```

Comment:
I've not used asserts in Locust before! How does the error output look? Since asserts like this one aren't unique to a task, I'm curious to know if we can trace an error back to the task that it originated from?

Reply:
These will print in the CLI so not quite useful for reporting but they do report as failures.